### DISCUSSION TITLE
Any example of cpp2 calling c++ code?

### DISCUSSION THREAD

**Discussion by felixf4xu** [hsutter/cppfront#977] [Q&A] [answered]
for example, if I have a c++ class, how can I initialize the class (`new` ?) and then call its member functions in `cpp2`?

**Comment by hsutter** [comment, marked as answer]
Welcome!

A class is a class regardless which syntax is used to author it. So you just use it as usual, though `new` is now a function call rather than a built-in language feature.

See in particular: [hello, world!](https://hsutter.github.io/cppfront/welcome/hello-world/) and [objects, initialization, and memory](https://hsutter.github.io/cppfront/cpp2/objects/).

**Comment by felixf4xu** [comment]
Thanks for the link of the document.

and I also found a perfect example code at https://github.com/hsutter/cppfront/blob/main/regression-tests/mixed-type-safety-1.cpp2

where `Circle` is a cpp1 class
```cpp
class Shape { public: virtual ~Shape() { } };
class Circle : public Shape { };
```
and then 
```cpp2
c := new<Circle>();     // safe by construction
```
here `new<>()` is a **cpp2** template function and it's transcompiled into some kind of `std::make_unique` in cpp1.
